### PR TITLE
Added a feature to load the server build details [#578]

### PR DESCRIPTION
**Pull request by mcpierce**
## Status
**READY**

## Migrations
NO

## Description
 * Added the frontend feature to retrieve the details.
 * Refactored the REST APIs and repackaged the classes.
 * Added the page to display the build details.

### CODE REVIEW

File: comixed-web/src/app/effects/build-details.effects.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    );
+  });
+
+  constructor(
+    private logger: LoggerService,
+    private actions$: Actions,
+    private buildDetailsService: BuildDetailsService,
+    private alertService: AlertService,
+    private translateService: TranslateService
+  ) {}
```
Comment (BRUCELLA2) [inline comixed-web/src/app/effects/build-details.effects.ts:71]: Is it voluntary to place the builder at the end of the class?
In the other effects.ts files the constructor is placed at the beginning of the class. This is consistent with a field -> constructor -> method structure. If this is what you want, you will have to correct the other classes in the course of time.

**Review by BRUCELLA2 — CHANGES_REQUESTED** [review]
it's ok just a question and we can merge

File: comixed-web/src/app/effects/build-details.effects.ts (hunk 1 above)
Comment (mcpierce) [reply comixed-web/src/app/effects/build-details.effects.ts:71]: I believe that's a side-effect of my IntelliJ setup and how it cleaned up the code. The constructor is the only pure method in the effect source module while the effect methods are generated by method calls to createEffect(). So IJ sorted those method calls above the constructor call. It's not a requirement for the code.

Comment (BRUCELLA2) [reply comixed-web/src/app/effects/build-details.effects.ts:71]: Okay, that's good then.

**Review by BRUCELLA2 — APPROVED** [review]

**Pull request merged**